src/Riok.Mapperly/Descriptors/Mappings/ExistingTarget/EnsureCapacityBuilder.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Riok.Mapperly.Helpers;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+using static Riok.Mapperly.Emit.SyntaxFactoryHelper;
+
+namespace Riok.Mapperly.Descriptors.Mappings.ExistingTarget;
+
+public class EnsureCapacityInfo
+{
+    public string? SourceAccessor { get; set; }
+    public string TargetAccessor { get; set; } = null!;
+    public TypeSyntax? CollectionType { get; set; }
```

Comment:
I'd stick to the symbols here instead of using the syntax and just build the typesyntax in the build method, if it is required there.